lib/fastlane/plugin/wpmreleasetoolkit/actions/common/create_branch.rb

```diff
@@ -0,0 +1,45 @@
+require 'fastlane/action'
+require_relative '../../helper/git_helper'
+module Fastlane
+  module Actions
+    class CreateBranchAction < Action
+      def self.run(params)
+        branch_name = params[:branch_name]
+        from = params[:from]
+
+        Fastlane::Helper::GitHelper.create_branch(branch_name, from: from)
+      end
+
+      def self.description
+        'Create a new branch named `branch_name`, cutting it from branch/commit/tag `from`'
+      end
+
+      def self.details
+        'If the branch with that name already exists, it will instead switch to it and pull new commits.'
+      end
```

Comment:
This behavior is confusing to me, especially since running `create_branch(branch_name: 'feature/abc', from: 'feature/def')` will end up just switching to `feature/abc` if it already exists… even if that existing branch was not cut from `feature/def`. So this would be misleading.

Again, I feel like it's the responsibility of the apps' `Fastfile` to checkout the right branch (e.g. `trunk`) before creating the new branch from the current commit.

---

What I'd suggest instead would be to have a `git_switch` action with a `branch` parameter, which will switch to an existing branch and `UI.user_error!` if it does not exist. Then have a `git_create_branch` action, maybe with an optional parameter `switch_after_create` (defaulting to `true`) which will return `true` if the branch was created and `false` if it already existed.

Then when one needs to cut a branch B from a branch A, they would call `git_switch(branch: A)` first, then `git_create_branch(name: B)`.